### DISCUSSION TITLE
Completion popop not showing all completion items.

### DISCUSSION THREAD

**Discussion by cronyakatsuki** [minad/corfu#392] [Q&A] [answered]
I have recently start setting up emacs and decided to go with corfu for my autocompletion, but there is an issue of the completion popup not showing all items correcty as seen in the next 3 screenshots.

> first screenshot

![pic-selected-231125-1453-06](https://github.com/minad/corfu/assets/64900606/6936cc9a-bb74-4045-9f6d-d8e5658998e5)

In this screenshot we can see that it just barelly shows the 9.th item.

> second screenshot

![pic-selected-231125-1458-11](https://github.com/minad/corfu/assets/64900606/9f027a28-6297-47e7-9375-e9a967f5af1a)

In this one I have selected the last item in the popup, but as seen it doesn't show in the popup.

> third screenshot

![pic-selected-231125-1500-03](https://github.com/minad/corfu/assets/64900606/4304ca46-83e3-476b-bf88-95ae54df2a93)

In this one I have selected the item before the last one, and as seen it shows itself just a little bit in the popup as selected.

This is related config for corfu and the popup:
``` emacs-lisp
(use-package corfu
  ;; Optional customizations
  :custom
  ;; Works with `indent-for-tab-command'. Make sure tab doesn't indent when you
  ;; want to perform completion
  (tab-always-indent 'complete)
  (completion-cycle-threshold nil)
  (corfu-cycle t)
  (corfu-auto t)
  (corfu-auto-prefix 2)
  (corfu-auto-delay 0.1)
  (corfu-min-width 60)
  (corfu-popupinfo-delay '(0.5 . 0.2))
  (corfu-preview-current 'insert)
  (corfu-preselect 'prompt)
  (corfu-on-exact-match nil)

  ;; Optionally use TAB for cycling, default is `corfu-complete'.
  :bind (:map corfu-map
              ("TAB"        . corfu-next)
              ([tab]        . corfu-next)
              ("S-TAB"      . corfu-previous)
              ([backtab]    . corfu-previous)
              ("S-<return>" . corfu-insert)
              ("RET"        . corfu-insert))
  :init
  (corfu-history-mode)
  (corfu-popupinfo-mode)
  (global-corfu-mode)
  :config
  (add-hook 'eshell-mode-hook
            (lambda () (setq-local corfu-quit-at-boundary t
                              corfu-quit-no-match t
                              corfu-auto nil)
              (corfu-mode))))
```

My system is a gentoo system, with emacs compiled with this flags: `X acl dynamic-loading gif gmp gtk gui imagemagick inotify jit jpeg json libxml2 png sound ssl svg threads tiff tree-sitter webp xft xpm zlib`

Is there any fix for this? Maybe a problem with emacs itself and not corfu? I know it's very possible this is just an emacs issue with a simple setting but I wasn't able to find what could possibly be the fix.

**Comment by minad** [comment]
Hi, it is likely that the candidates are not shown fully because of the icons which are too large, or because of some other font-related settings. Did you try without icons?

**Reply by cronyakatsuki** [reply]
Can confirm it's an issue becaue of icons. Removed kind icons and the candidates now show correctly.

**Reply by cronyakatsuki** [reply, marked as answer]
Will report the issue to the kind-icon maintainer.
